server/hydrogen-render/render-hydrogen-to-string-unsafe.js

```diff
@@ -102,19 +98,43 @@ async function _renderHydrogenToStringUnsafe(renderOptions) {
 
   const vmRenderScriptFilePath = renderOptions.vmRenderScriptFilePath;
   const hydrogenRenderScriptCode = await readFile(vmRenderScriptFilePath, 'utf8');
-  const hydrogenRenderScript = new vm.Script(hydrogenRenderScriptCode, {
-    filename: path.basename(vmRenderScriptFilePath),
+  const hydrogenRenderModule = new vm.SourceTextModule(hydrogenRenderScriptCode, {
+    identifier: path.basename(vmRenderScriptFilePath),
+    context: vmContext,
+  });
+  // TODO: This is crazy compared to the normal `vm.Script(...)` API and this doesn't
+  // work at the moment because it only handles ESM imports or exports, idk. Too much BS
+  // if I just want the behavior that Node.js already provides by default.
+  //
+  // And the only reason we need to use `vm.SourceTextModule` is because we need see
+  // `SyntaxError: Cannot use import statement outside a module` in the script we try to
+  // run with `vm.Script(...)` and `vm.runInContext(vmContext);`.
+  //
+  // XXX: This is crazy to handle on our own!
+  // https://github.com/nodejs/node/issues/31234 Also see
+  // https://github.com/nodejs/node/issues/35848
+  await hydrogenRenderModule.link((specifier, referencingModule) => {
+    return new Promise(async function (resolve, reject) {
+      const mod = await import(specifier);
+      resolve(
+        new vm.SyntheticModule(['default'], function () {
+          this.setExport('default', mod.default);
+        })
+      );
+    });
   });
-  // Note: The VM does not exit after the result is returned here and is why
-  // this should be run in a `child_process` that we can exit.
-  const vmResult = hydrogenRenderScript.runInContext(vmContext);
-  // Wait for everything to render
-  // (waiting on the promise returned from the VM render script)
-  await vmResult;
+  await hydrogenRenderModule.evaluate();
```

Comment:
This area will be the downfall of this migration 😢

With the previous `new vm.Script(code).runInContext(vmContext)`, we see `SyntaxError: Cannot use import statement outside a module`

Which prompted an update to `new vm.SourceTextModule(...) ...` but this [API surface](https://nodejs.org/docs/latest-v18.x/api/vm.html#class-vmmodule) is crazy and expects to implement our own `linker` when I just want default Node.js behavior.

 - https://github.com/nodejs/node/issues/31234
 - https://github.com/nodejs/node/issues/35848
 - https://github.com/jestjs/jest/blob/6e95bdcbf9e5c2e354f277d0db6e4dff1ad2c43f/packages/jest-runtime/src/index.ts#L474-L513
 
The `new vm.Script(...)` API should handle all of the complexity of whether I used `import` or `require` syntax and just load dependencies (sane defaults please).

Reply:
(giving up for now ⏩)